Netezos.Tests/Forging/LocalForgingTest.cs

```diff
@@ -76,5 +76,68 @@ public async Task CompareToFile()
                 }
             }
         }
+
+        [Fact]
+        public async Task CompareUnforgedFromFile()
+        {
+            var localForge = new LocalForge();
+            var options = new JsonSerializerOptions
+            {
+                DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull,
+                MaxDepth = 1024,
+                NumberHandling = JsonNumberHandling.AllowReadingFromString
+            };
+            options.Converters.Add(new JsonStringEnumConverter(JsonNamingPolicy.CamelCase));
+            options.Converters.Add(new DateTimUtcTimezoneAppender());
+
+            var jsonComparer = new JsonElementComparer(true, true);
+            var basePath = @"../../../Forging/operations";
+            var directories = Directory.GetDirectories(basePath);
+
+            Func<object, JsonElement> toJsonElement = (o) =>
+            {
+                // JSON serialize the unforged manager operation for the purpose of deserializing as JsonElement for comparison
+                var serMop = JsonSerializer.Serialize(o, o.GetType(), options);
+                // Deserialize the unforged manager operation to JsonElement for comparison
+                return (JsonElement)DJson.Parse(serMop);
+            };
+
+            foreach (var directory in directories)
+            {
+                var json = DJson.Read($"{directory}/unsigned.json", options);
+
+                // Parse the op bytes
+                var opBytes = Hex.Parse(File.ReadAllText($"{directory}/forged.hex"));
+
+                switch ((int)json.contents.count)
+                {
+                    case int n when (n > 1):
+                        // Unforge the manager op bytes
+                        var mops = await localForge.UnforgeOperationGroupAsync(opBytes);
+
+                        // Serialize/deserialize each operation for the purpose of conversion to JsonElement for comparison
+                        var deserMops = mops.Item2.Select(toJsonElement);
+
+                        // Assert branch
+                        Assert.Equal(json.branch, mops.Item1);
+                        // Assert equivalent JSON operations
+                        Assert.True(((JsonElement)json.contents).EnumerateArray().SequenceEqual(deserMops, jsonComparer));
+                        break;
+
+                    case int n when (n == 1):
+                        // Unforge the op bytes
+                        var op = await localForge.UnforgeOperationAsync(opBytes);
+
+                        // Assert branch
+                        Assert.Equal(json.branch, op.Item1);
+                        // Assert equivalent JSON operations
+                        Assert.True(jsonComparer.Equals((JsonElement)json.contents[0], toJsonElement(op.Item2)));
+                        break;
+
+                    default:
+                        throw new ArgumentException();
```

Comment:
Can anyone think of a better way to discern between group and single operations simply from the forged bytes? As of now, you can see the tests rely on the baseline JSON to determine if it should be unforged to a group (ManagerOperation) or a single (Operation). I'm not sure how this would be used in practice and if that's an acceptable approach.

Reply:
See https://github.com/baking-bad/netezos/pull/25/files#r579675794